queue/queueservice.js

```diff
@@ -288,8 +284,24 @@ class QueueService {
            "Expected provisionerId to be an identifier");
     assert(/^[A-Za-z0-9_-]{1,22}$/.test(workerType),
            "Expected workerType to be an identifier");
+
+    // Hash identifier to 24 characters
+    let hashId = (id) => {
+      var h = crypto.createHash('sha256').update(id).digest();
+      return base32.encode(h.slice(0, 15)).toString('utf-8').toLowerCase();
+    };
+
     // Construct queue name
     var name = [
+      this.prefix,            // prefix all queues
+      hashId(provisionerId),  // hash of provisionerId
+      hashId(workerType),     // hash of workerType
+      '1'                     // priority, currently just hardcoded to 1
```

Comment:
The `priority` part of the name is just so we create new queues in the future...

Ie. when we decide to add priority support we add a queue where the priority number is `2` and `3`, and then we have 3 queue names... Hence, we can implement priority...